src/jit/gentree.cpp

```diff
@@ -611,6 +611,60 @@ void Compiler::fgWalkAllTreesPre(fgWalkPreFn* visitor, void* pCallBackData)
 }
 
 // ------------------------------------------------------------------------------------------
+// gtCheckReg: Checks that the register has either been set, or is still set to the "no register
+// assignment" value, depending upon whether it fits any of the special cases for multi-reg call nodes.
+//
+// Arguments:
+//     compiler  -  compiler instance
+//
+// Return Value:
+//     None
+//
+// Notes:
+//     This also sets the gtRegTag to GT_REGTAG_REG.
+//
+void GenTree::gtCheckReg()
+{
+#ifdef DEBUG
+    assert((gtRegTag != GT_REGTAG_NONE) || (gtRegNum == REG_NA));
+#if CPU_LONG_USES_REGPAIR
+    assert(!"gtSetRegDefault not supported for REGPAIR");
+#endif // CPU_LONG_USES_REGPAIR
+    if (gtRegTag == GT_REGTAG_NONE)
+    {
+        gtRegTag = GT_REGTAG_REG;
+
+// Also check multi-reg state
+#if FEATURE_MULTIREG_RET
+        if (IsCall())
+        {
+            for (unsigned i = 0; i < MAX_RET_REG_COUNT - 1; ++i)
+            {
+                assert(this->AsCall()->gtOtherRegs[i] == REG_NA);
+            }
+        }
+        else if (IsCopyOrReload())
+        {
+            for (unsigned i = 0; i < MAX_RET_REG_COUNT - 1; ++i)
+            {
+                assert(this->AsCall()->gtOtherRegs[i] == REG_NA);
+            }
+        }
+#endif // FEATURE_MULTIREG_RET
```

Comment:
This looks the same as above; can they be merged? But if `IsCopyOrReload` is true, don't you need `this->AsCopyOrReload()`, not `this->AsCall()`?